src/player.cpp

```diff
@@ -3876,13 +3876,17 @@ void player::practice( const skill_id &id, int amount, int cap, bool suppress_wa
                      skill_name );
         }
 
-        int chance_to_drop = focus_pool;
-        focus_pool -= chance_to_drop / 100;
+        // Calculate focus chance to drop sinusoidally
+        int chance_to_drop = roll_remainder( static_cast<double>( 50 * ( std::cos( M_PI * ( (
+                focus_pool % 100 ) + 100 ) / 100 ) + 1 ) ) );
+        focus_pool -= focus_pool / 100;
+
         // Apex Predators don't think about much other than killing.
         // They don't lose Focus when practicing combat skills.
-        if( ( rng( 1, 100 ) <= ( chance_to_drop % 100 ) ) && ( !( has_trait_flag( "PRED4" ) &&
-                skill.is_combat_skill() ) ) ) {
-            focus_pool--;
+        if( ( !( has_trait_flag( "PRED4" ) && skill.is_combat_skill() ) ) ) {
+            if( ( rng( 1, 100 ) <= chance_to_drop ) ) {
+                focus_pool--;
+            }
         }
```

Comment:
This is basically the same code in two places.  Can you factor it out into a function?